src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -191,7 +178,8 @@ protected function getSupportedAttributes()
      */
     protected function isGranted($attribute, $object, $user = null)
     {
-        return false;
+        // forces isGranted() or voteOnAttribute() to be overridden
+        throw new \BadMethodCallException(sprintf('You must override either the voteOnAttribute() or isGranted() (deprecated) method in "%s"', get_class($this)));
```

Comment:
we can only mention `voteOnAttribute()` here, can't we? No need to talk about overriding a method that shouldn't be used anymore